File: LLD NodeJs/p1_oops/c7_composition/association/Teacher.js
/**
 * Teacher class for demonstrating Association
 * Teacher exists independently and can be shared by multiple Students
 */
class Teacher {
    #name;
    #subject;

    constructor(name, subject) {
        this.#name = name;
        this.#subject = subject;
    }

    getName() {
        return this.#name;
    }

    getSubject() {
        return this.#subject;
    }

    teach() {
        console.log("Teacher " + this.#name + " is teaching " + this.#subject);
    }
}

module.exports = Teacher;

File: LLD/src/main/java/org/example/p1_oops/c7_composition/association/Teacher.java
package org.example.p1_oops.c7_composition.association;

public class Teacher {
    private String name;
    private String subject;

    public Teacher(String name, String subject) {
        this.name = name;
        this.subject = subject;
    }

    public String getName() {
        return name;
    }

    public String getSubject() {
        return subject;
    }

    public void teach() {
        System.out.println("Teacher " + name + " is teaching " + subject);
    }
}
